OneSignalSDK/unittest/src/test/java/com/test/onesignal/TestHelpers.java

```diff
@@ -499,26 +503,97 @@ public static void assertMainThread() {
+              (JobScheduler) ApplicationProvider.getApplicationContext().getSystemService(Context.JOB_SCHEDULER_SERVICE);
+      if (jobScheduler != null) {
+         List<JobInfo> jobs = jobScheduler.getAllPendingJobs();
+         for (JobInfo jobInfo : jobs) {
+            stopJob(jobInfo);
+            threadAndTaskWait();
+         }
+      }
+   }
    public static @Nullable JobInfo getNextJob() {
+      return getJob(0);
+   }
+
+   private static @Nullable JobInfo getJob(int index) {
       JobScheduler jobScheduler =
-         (JobScheduler)ApplicationProvider.getApplicationContext().getSystemService(Context.JOB_SCHEDULER_SERVICE);
+              (JobScheduler)ApplicationProvider.getApplicationContext().getSystemService(Context.JOB_SCHEDULER_SERVICE);
       List<JobInfo> jobs = jobScheduler.getAllPendingJobs();
-      if (jobs.size() == 0)
+      if (jobs.size() == 0 || jobs.size() <= index)
          return null;
-      return jobs.get(0);
+      else
+         return jobs.get(index);
    }
 
    public static void runNextJob() {
       try {
          Class jobClass = Class.forName(getNextJob().getService().getClassName());
-         JobService jobService = (JobService)Robolectric.buildService(jobClass).create().get();
-         jobService.onStartJob(null);
+         runJob(jobClass);
       } catch (ClassNotFoundException e) {
          e.printStackTrace();
       }
    }
 
+   public static void runJob(Class jobClass) {
+      JobService jobService = (JobService) Robolectric.buildService(jobClass).create().get();
+      jobService.onStartJob(null);
+   }
+
+   public static void stopJob(JobInfo jobInfo) {
+      try {
+         Class jobClass = Class.forName(jobInfo.getService().getClassName());
+         JobService jobService = (JobService) Robolectric.buildService(jobClass).create().get();
+         jobService.jobFinished(null, false);
+      } catch (ClassNotFoundException e) {
+         e.printStackTrace();
+      }
+   }
+
+   public static void assertNumberOfServicesAvailable(int quantity) {
+      JobScheduler jobScheduler =
+              (JobScheduler)ApplicationProvider.getApplicationContext().getSystemService(Context.JOB_SCHEDULER_SERVICE);
+      List<JobInfo> jobs = jobScheduler.getAllPendingJobs();
+      assertEquals(quantity, jobs.size());
+   }
+
+   public static void assertAndRunNextJob(Class jobClass) {
+      assertNumberOfServicesAvailable(1);
+      assertNextJob(jobClass);
+      runNextJob();
+   }
+
+   public static void assertAndRunJobAtIndex(Class jobClass, int index) {
+      assertNextJob(jobClass, index);
+      runJob(jobClass);
+   }
+
+   public static void assertNextJob(Class jobClass) {
+      assertNextJob(jobClass, 0);
+   }
+
+   public static void assertNextJob(Class jobClass, int index) {
+      assertEquals(jobClass.getName(), getJob(index).getService().getClassName());
+   }
+
+   public static void pauseActivity(ActivityController activityController) throws Exception {
```

Comment:
👍 Nice helper!

It was a lot of line changes, would be nice in the future if this it was its own commit. That way it is easier to review the other changes in smaller chunks.